src/AMSlib/wf/basedb.hpp

```diff
@@ -112,6 +112,8 @@ class BaseDB
                      std::vector<TypeValue*>& outputs) = 0;
 
   uint64_t getId() const { return id; }
+
+  virtual bool updateModel() { return false; }
```

Comment:
I am not sure managing that with a `bool` is enough. With this model, the first time we update it's easy, I set the `bool` to `true` but then for the second updates I have to set it to false once the first update went through.
We might have to store the latest model path used for example, so we know that we have to update the model if paths are different.

Note that maybe it's for me to implement and you just picked a simple way of testing the model update, if that's the case disregard that comment :)

Reply:
I agree, I thought we should re-iterate this. But we need to return something. 

Maybe we should do something among the lines of:
`model.getCurrentModel() != DB.mostRecentModel()`

We should reiterate this part once we have the mechanism in place to receive a new model.